### PR TITLE
feature(utility-panel): style toggle buttons

### PR DESCRIPTION
**Pull request by frankharkins**
## Changes

Styled the show/hide buttons for the utility panel

## Implementation details

- I couldn't work out how to get the `bx-btn` working, so I copied the code for the index toggle button, and pulled this into its own component (`SidebarToggleButton`)

## Screenshots

Here's the thing working:

https://user-images.githubusercontent.com/36071638/205626366-128166ff-ff9c-4814-bde7-416bf3199966.mov

Question: what's the intended behaviour for thin windows? Both the content and sidebar have a minimum width, should the sidebar overlap the content?

Screenshot 2022-12-05 at 11 23 15

### CODE REVIEW

File: components/SidebarToggleButton.vue [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<template>
+  <button type="button" class="sidebar-toggle-button">
```
Comment (yoshiemuranaka) [inline components/SidebarToggleButton.vue:2]: >I couldn't work out how to get the bx-btn working

Was there something specific you got stuck on? the `<bx-btn>` component is a carbon [web component](https://web-components.carbondesignsystem.com/?path=/story/components-button--default). [Docs](https://web-components.carbondesignsystem.com/?path=/docs/components-button--default). 

Wondering if any of these styles could be removed/cleaned up if we used the carbon button?

**Review by yoshiemuranaka — COMMENTED** [review]
these changes look good to me! just left a comment about using the bx-btn web component

**Comment by techtolentino** [comment]
@frankharkins 

> Question: what's the intended behaviour for thin windows? Both the content and sidebar have a minimum width, should the sidebar overlap the content?

Yes, overlapping is fine for now, mainly because if a user is on a tablet device, the page loads w/ the utility panel hidden by default.

File: components/SidebarToggleButton.vue (hunk 1 above)
Comment (techtolentino) [reply components/SidebarToggleButton.vue:2]: same question as Yoshie..

let us know if we can help w/ overriding styles, if that's what the issue was.

💪🏼

Comment (frankharkins) [reply components/SidebarToggleButton.vue]: Thanks! @yoshiemuranaka helped me sort it out via slack

**Comment by abdonrd** [comment]
@frankharkins could you rebase the branch? Now I see a lot of external commits:

Screenshot 2022-12-07 at 9 04 39 AM

File: components/SidebarToggleButton.vue [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+.toggle-button {
+  --cds-interactive-01: #{carbon.$cool-gray-10};
+  --cds-hover-primary: #{carbon.$cool-gray-10};
+  --cds-active-primary: #{carbon.$cool-gray-10};
+  --cds-focus: #{carbon.$cool-gray-10};
+  --cds-text-04: #{carbon.$purple-70};
+
+  max-width: none;
+
+  &::part(button) {
```
Comment (techtolentino) [inline components/SidebarToggleButton.vue:19]: Nice 🚀

**Review by techtolentino — APPROVED** [review]
Looks great @frankharkins! 

Just had a couple of non-blocking questions... 

👏🏼

File: components/UtilityPanel.vue [hunk 3]
```diff
@@ -23,6 +32,20 @@ function hide() {
 @use '~/assets/scss/carbon.scss';
 
 .utility-panel {
+  display: flex;
+  flex-flow: column;
+  height: 100%;
   border-left: 1px solid carbon.$cool-gray-10;
+  background: white;
```
[the remark this answers is not in the document]
Comment (frankharkins) [reply components/UtilityPanel.vue]: Yes, done!

File: pages/course/[course]/[unit].vue [hunk 4]
```diff
@@ -4,39 +4,37 @@
...
         v-if="noSidebarIsShown"
-        class="sidebar-mobile-button"
         type="button"
         @click="toggleSidebar"
       >
-        <span>Show index</span>
-        <IconTogglePanelLeft class="sidebar-mobile-button__icon" />
-      </button>
-      <bx-btn
+        Index
```
[the remark this answers is not in the document]
Comment (frankharkins) [reply pages/course/[course]/[unit].vue:12]: Ahh yes sorry I should have called this out in the PR description; I removed 'show' to match the 'glossary' button. I think it's cleaner like that but would be best to have a conversation about it.

[the remark this answers is not in the document]
Comment (techtolentino) [reply pages/course/[course]/[unit].vue:12]: I'm good w/ that... We can re-add the "Show" once the "Glossary" button becomes more than just the glossary.

File: package.json [hunk 5]
```diff
@@ -53,7 +53,10 @@
     "vitest": "0.24.3"
   },
   "eslintConfig": {
-    "extends": "./packages/eslint-config"
+    "extends": "./packages/eslint-config",
+    "rules": {
+      "vue/no-deprecated-slot-attribute": "off"
```
Comment (yoshiemuranaka) [inline package.json:58]: Ok for me–I guess we should be mindful not to use the deprecated `slot` syntax. Also maybe can clean up the lint ignore pattern in a separate PR 
https://github.com/Qiskit/saiba/search?q=eslint-disable-next-line+vue%2Fno-deprecated-slot-attribute

**Review by yoshiemuranaka — APPROVED** [review]
Looks great, thank you!

File: pages/course/[course]/[unit].vue [hunk 6]
```diff
@@ -4,39 +4,37 @@
       <CourseSidebar />
     </template>
     <template #sidebar-mobile>
-      <button
+      <SidebarToggleButton
         v-if="noSidebarIsShown"
-        class="sidebar-mobile-button"
         type="button"
```
Comment (abdonrd) [inline pages/course/[course]/[unit].vue:9]: We can remove it, right?

File: pages/course/[course]/[unit].vue [hunk 7]
```diff
@@ -4,39 +4,37 @@
...
-        <IconTogglePanelLeft class="sidebar-mobile-button__icon" />
-      </button>
-      <bx-btn
+        Index
+        <IconTogglePanelLeft slot="icon" />
+      </SidebarToggleButton>
+      <SidebarToggleButton
         v-if="noSidebarIsShown"
-        class="sidebar-mobile-button"
         type="button"
```
Comment (abdonrd) [inline pages/course/[course]/[unit].vue:17]: We can remove it, right?

File: pages/course/[course]/[unit].vue (hunk 6 above)
Comment (frankharkins) [reply pages/course/[course]/[unit].vue:9]: Seems so...

File: package.json (hunk 5 above)
Comment (frankharkins) [reply package.json:58]: Thanks! I've followed up in https://github.com/Qiskit/saiba/pull/389

File: pages/course/[course]/[unit].vue (hunk 6 above)
Comment (abdonrd) [reply pages/course/[course]/[unit].vue:9]: For that we need to accept type as a prop in the `SidebarToggleButton`.
And then pass it to the `bx-btn` (if it accepts it).

Comment (frankharkins) [reply pages/course/[course]/[unit].vue:9]: Interesting, I didn't realise that's what that attribute did. I don't think `SidebarToggleButton` should be used in any forms, so is it ok to just fix it as 'button'? https://github.com/Qiskit/saiba/pull/392/files

**Pull request merged**